src/main/scala/sigmastate/lang/SigmaParser.scala

```diff
@@ -59,18 +59,19 @@ object SigmaParser extends Exprs with Types with Core {
     case "||" => OR(l.asValue[SBoolean.type], r.asValue[SBoolean.type])
     case "&&" => AND(l.asValue[SBoolean.type], r.asValue[SBoolean.type])
     case "==" => EQ(l, r)
+    case "!=" => NEQ(l, r)
     case ">=" => GE(l, r)
     case ">"  => GT(l, r)
     case "<=" => LE(l, r)
     case "<"  => LT(l, r)
-    case "+"  => Plus(l.asValue[SLong.type], r.asValue[SLong.type])
-    case "-"  => Minus(l.asValue[SLong.type], r.asValue[SLong.type])
+    case "+"  => StdSigmaBuilder.Plus(l.asValue[SLong.type], r.asValue[SLong.type])
```

Comment:
@greenhat Let's pass the builder as a parameter to the parser. So we can configure parsing with different builders. This is one of the reasons to introduce builders in a first place. :)

Reply:
@ergomorphic Damn it, that makes so much sense. :)